### PR TITLE
Update how-to-handle-multiple-events-using-event-properties.md

### PR DESCRIPTION
**Pull request by goyzhang**
Improve the description for the issue

#23356

### CODE REVIEW

File: docs/standard/events/how-to-handle-multiple-events-using-event-properties.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ ms.assetid: 30047cba-e2fd-41c6-b9ca-2ad7a49003db
 ---
 # How to: Handle Multiple Events Using Event Properties
 
-To use event properties, you define the event properties in the class that raises the events, and then set the delegates for the event properties in classes that handle the events. To implement multiple event properties in a class, the class must internally store and maintain the delegate defined for each event. A typical approach is to implement a delegate collection that is indexed by an event key.  
+To use event properties, you define the event properties in the class that raises the events, and then set the delegates for the event properties in classes that handle the events. To implement multiple event properties in a class, the class should internally store and maintain the delegate defined for each event. A typical approach is to implement a delegate collection that is indexed by an event key. Because each field-like-event requires a reference backing-field. And when the number of events increases, a lot of fields will be created even they are not subscribed. This can introduce unnecessary allocations when only a few events are subscribed.
```
Comment (IEvangelist) [inline docs/standard/events/how-to-handle-multiple-events-using-event-properties.md:18]:
```suggestion
To use event properties, you define the event properties in the class that raises the events, and then set the delegates for the event properties in classes that handle the events. To implement multiple event properties in a class, the class should internally store and maintain the delegate defined for each event.

As the number of events increases, so too does the number of backing-field references. To help avoid unnecessary allocations an approach is to implement a delegate collection that is indexed by an event key.
```

**Review by IEvangelist — COMMENTED** [review]
How do you feel about this wording instead?

File: docs/standard/events/how-to-handle-multiple-events-using-event-properties.md (hunk 1 above)
Comment (goyzhang) [reply docs/standard/events/how-to-handle-multiple-events-using-event-properties.md:18]: "As the number of events increases" is imprecise, as you can see from the code example, even with self-defined add/remove accessors the number of events still increases anyway. The issue only happens when you rely on the complier's syntax sugar, a "field-like-event", which will generate a backing-field:
`public event EventHandler<EventArgs> MyEvent;`

Comment (IEvangelist) [reply docs/standard/events/how-to-handle-multiple-events-using-event-properties.md:18]: I updated my suggestion, please have another look - and let me know what you think.

Comment (goyzhang) [reply docs/standard/events/how-to-handle-multiple-events-using-event-properties.md:18]: HI, IMO it's generally good😊, but why "must" again? As you can see in you example `Progress`  and `AnotherProgress` generated backing-field, they still work but suffer from the issue when the number of events increases, but `ManualProgress` doesn't.

Comment (goyzhang) [reply docs/standard/events/how-to-handle-multiple-events-using-event-properties.md]: I've updated the doc according to your suggestion. Take a look

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**